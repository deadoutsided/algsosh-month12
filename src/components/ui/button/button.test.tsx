import React from "react";
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react'

import { Button } from "./button";

describe('button appearance', () => {
  it('hollow button', () => {
    const tree = renderer.create(<Button/>).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('button with text', () => {
    const tree = renderer.create(<Button text='Hello World!' />).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('blocked button', () => {
    const tree = renderer.create(<Button disabled={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('Button with Loader', () => {
    const tree = renderer.create(<Button isLoader={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('button click causes correct allert', () => {
    window.alert = jest.fn();
    render(<Button text='рецепт пельменей' onClick={() => {alert('пельмени')}} />)
    const btn = screen.getByText('рецепт пельменей');
    fireEvent.click(btn);
    expect(window.alert).toHaveBeenCalledWith('пельмени');
  })
})