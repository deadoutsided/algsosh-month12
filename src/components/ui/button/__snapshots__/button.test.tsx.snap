// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button appearance Button with Loader 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`button appearance blocked button 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`button appearance button with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Hello World!
  </p>
</button>
`;

exports[`button appearance hollow button 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
